plugin/pkg/shoot/validator/admission.go

```diff
@@ -447,6 +508,41 @@ func validateAWS(c *validationContext) field.ErrorList {
 	return allErrs
 }
 
+// validateKubeletWithNodeCIDRMaskSize validates if the maxPod settings on the worker pools conflict with the kubecontroller NodeCIDRMaskSize
+// chooses a suitable NodeCIDRMaskSize if none is configured
+func validateKubeletWithNodeCIDRMaskSize(maxPod int32, kubeControllerManagerConfig *garden.KubeControllerManagerConfig, operation admission.Operation) (*int, error) {
```

Comment:
Why does of all of this happen here in the admission plugin and not in the static validation? It's not needed to be here. Please move it to `pkg/apis/garden/validation`.